### PR TITLE
Creating bot account and enabling intents guide

### PR DESCRIPTION
**Pull request by Stonedestroyer**
### Type

- [ ] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes
Adds getting started and intent guide, adapted from d.py docs as well.

### CODE REVIEW

File: docs/bot_application_guide.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        :alt: The bot tab in the application page.
+
+5. Scroll down to the "Privileged Gateway Intents" section and both privileged intents.
+
+    .. image:: /.resources/bot-guide/discord_privileged_intents.png
+        :alt: The privileged gateway intents selector.
+
+.. warning::
+
+    Red bots with over 100 servers require `bot verification <https://support.discord.com/hc/en-us/articles/360040720412>`_ and is not covered in this guide.
```
Comment (Jackenmen) [inline docs/bot_application_guide.rst]: I think you wanted to use `which` here:
```suggestion
    Red bots with over 100 servers require `bot verification <https://support.discord.com/hc/en-us/articles/360040720412>`_ which is not covered in this guide.
```

File: docs/bot_application_guide.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    .. image::  /.resources/bot-guide/discord_create_app_form.png
+        :alt: The new application form filled in.
+
+5. Create a Bot User by navigating to the "Bot" tab and clicking "Add Bot".
+
+    - Click "Yes, do it!" to continue.
+
+    .. image::  /.resources/bot-guide/discord_create_bot_user.png
+        :alt: The Add Bot button.
+6. If you want your bot to be invitable by others tick the **Public Bot**, keeping it unticked will only allow you to add your bot to servers that you have permissions in.
```
Comment (Jackenmen) [inline docs/bot_application_guide.rst]: Rewording of the first partto remove "invitable" which is not really a word
+ some rewording for the second part that I think might be a bit easier to read
```suggestion
6. If you want others to be able to invite your bot tick the **Public Bot**. Keeping it unticked will prevent others from inviting your bot to their servers and only you will be able to add the bot to servers (provided that you have needed permissions in the server you want to add the bot to).
```

File: docs/bot_application_guide.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+5. Scroll down to the "Privileged Gateway Intents" section and both privileged intents.
+
+    .. image:: /.resources/bot-guide/discord_privileged_intents.png
+        :alt: The privileged gateway intents selector.
+
+.. warning::
+
+    Red bots with over 100 servers require `bot verification <https://support.discord.com/hc/en-us/articles/360040720412>`_ and is not covered in this guide.
+
+*Parts of this guide has been adapted from* `discord.py intro <https://discordpy.readthedocs.io/en/stable/discord.html#discord-intro>`_ *and* `discord.py privileged intents <https://discordpy.readthedocs.io/en/stable/intents.html#privileged-intents>`_.
```
Comment (Jackenmen) [inline docs/bot_application_guide.rst]: Wrong form
```suggestion
*Parts of this guide have been adapted from* `discord.py intro <https://discordpy.readthedocs.io/en/stable/discord.html#discord-intro>`_ *and* `discord.py privileged intents <https://discordpy.readthedocs.io/en/stable/intents.html#privileged-intents>`_.
```

File: docs/bot_application_guide.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+1. Make sure you're logged on to the `Discord website <https://discord.com>`_.
+2. Navigate to the `application page <https://discord.com/developers/applications>`_
+3. Click on the bot you want to enable privileged intents for.
+4. Navigate to the bot tab on the left side of the screen.
+
+    .. image:: /.resources/bot-guide/discord_bot_tab.png
+        :alt: The bot tab in the application page.
+
+5. Scroll down to the "Privileged Gateway Intents" section and both privileged intents.
```
Comment (Jackenmen) [inline docs/bot_application_guide.rst]: You missed a word here:
```suggestion
5. Scroll down to the "Privileged Gateway Intents" section and enable both privileged intents.
```

Also, do you think we should say that the user needs to click the "Save Changes" button after this?

**Review by Jackenmen — CHANGES_REQUESTED** [review]

File: redbot/core/cli.py [hunk 5]
```diff
@@ -42,9 +42,8 @@ async def interactive_config(red, token_set, prefix_set, *, print_header=True):
     if not token_set:
         print(
             "Please enter a valid token.\n"
-            "You can find out how to obtain a token with this guide"
-            ' (section "Creating a Bot Account"):\n'
-            f"https://discordpy.readthedocs.io/en/v{discord_version}/discord.html#creating-a-bot-account"
+            "You can find out how to obtain a token with this guide:\n"
+            "https://docs.discord.red/en/stable/bot_application_guide#creating-a-bot-account"
```
Comment (Jackenmen) [inline redbot/core/cli.py]: I haven't looked too closely at this link at first, but looks like you missed `.html` extension here:
```suggestion
            "https://docs.discord.red/en/stable/bot_application_guide.html#creating-a-bot-account"
```

File: docs/bot_application_guide.rst (hunk 4 above)
Comment (Stonedestroyer) [reply docs/bot_application_guide.rst]: I think that could be a good idea as Discord is not obvious with the save.

Comment (Stonedestroyer) [reply docs/bot_application_guide.rst]: added

**Review by Jackenmen — APPROVED** [review]
Looks good to me, thanks sooooo much for helping me out with this ❤️

**Pull request merged**